talp03-sp

AsemikTEA opened the pull request:
připojení k db, nějaké základní rozdělení a zkouška funkčnosti

AsemikTEA committed:
talp03-sp

připojení k db, nějaké základní rozdělení a zkouška funkčnosti

AsemikTEA committed:
update 1

přihlašování, registrace, košík, z poloviny objednávky, editace zboží, filtrace podle kategorie

AsemikTEA commented:
Dobrý den, měl bych dotaz na práci s databází.

Nejdříve editace produktů. Když chci na straně `edit-product` uložit změny, tak mi vše projde, ale v databázi se nic nezmění. Myslím, že příkaz v `ProductsDB` metoda `updateProduct` mám napsaný správně, je stejný, který jsme si ukazovali na cvičení. nevíte čím by to mohlo být?

Druhá otázka je, jak bych měl uložit záznamy do tabulky `order-products`. Je nějaká možnost, jak získat `order_id` z příkazu pro uložení objednávky v `OrderDB`?

Pro snadnější pochopení druhého dotazu přikládám model databáze
![Relational database](https://github.com/nvbach91/4IZ278-2023-2024-LS/assets/139178681/df21d48c-2e07-4521-a3dd-d3a8112b80b4)

nvbach91 commented on www/talp03/sp/Controller/editProductPage.php:
```diff
@@ -0,0 +1,17 @@
+$product = $productDB->find('products', 'product_id', $productID)[0];
+
+if (!empty($_POST)) {
+    $productDB->updateProduct($_POST, $productID);
```
promenna `$productID` neexistuje v pripade submitu formulare

nvbach91 commented on www/talp03/sp/Model/OrderDB.php:
```diff
@@ -0,0 +1,15 @@
+
+class OrderDB extends Database {
+    
+    public function createOrder($order) {
```
createOrder bude delat:
- INSERT novy order (`user_id`, `date`, `total_price`, ...) a z tohoto INSERTu dostanete `order_id`
- INSERT polozky orderu s pouzitim noveho `order_id` ktery jste dostal z prvniho insertu

AsemikTEA committed:
update 2

edit privilegií, zprovoznění editu produktů, profil uživatele, vytvoření nového produktu

AsemikTEA committed:
update 2

oprava editu produktů, přidání nového produktu, editace privilegií, profil uživatele

AsemikTEA committed:
Update 3

wishlist, dodělání poslání objednávky, pokus o Oauth s githubem

AsemikTEA committed:
update 4

pagination, odstranění produktu, zabezpečení vstupních polí, optimistic lock

AsemikTEA commented:
Projekt je dostupný na adrese https://eso.vse.cz/~talp03/sp/View/index.php

AsemikTEA committed:
update 5

zaplacenéí/stornování objednávky, posílání mailů,

AsemikTEA committed:
Final Update

zprovoznění Oauth, nakonec s Googlem, původně jsem zkoušel github.

AsemikTEA committed:
Final Update 2

Github v původním commitu nějak nezaznamenal změny. Dodělání Oauth nakonec s googlem.

nvbach91 reviewed (changes requested):
na co jsou ty slozky `/Controller` `/Model` `/View` v korene projektu?

nvbach91 commented on www/talp03/sp/Model/DatabaseConnection.php:
```diff
@@ -0,0 +1,22 @@
+<?php
+
+const DB_DATABASE = 'sp_eshop';
```
tohle dejte do samostatneho souboru config.php nebo .env a chrante ho pomoci .htaccess

AsemikTEA replied:
Vyřešeno

nvbach91 commented on www/talp03/sp/View/cart.php:
```diff
@@ -0,0 +1,34 @@
+    <main>
+        <div class="container-cart">
+            <div class="products">
+                <?php foreach ($products as $product): ?>
```
proc se mi neinkrementuje pocet ale duplikuje se polozka?

nvbach91 commented on www/talp03/sp/Controller/makeOrder.php:
```diff
@@ -0,0 +1,61 @@
+require '../Model/ProductDB.php';
+require '../Model/UserDB.php';
+
+$priceSum = $_GET['price'];
```
jaktoze se mi podarilo objednat za 7 Kc?

![image](https://github.com/nvbach91/4IZ278-2023-2024-LS/assets/20724910/c585ce91-ea7e-492b-b67d-227dce47723a)

AsemikTEA replied:
problém předpokládám byl, že jsem cenu přenášel pomocí GET což bylo bezpečností riziko. Přesunul jsem to do SESSION

nvbach91 commented on www/talp03/sp/View/wishlist.php:
```diff
@@ -0,0 +1,29 @@
+                        <div class="product-body">
+                            <h5 class="product-name"><?php echo $product['name']; ?></h5>
+                            <p class="product-description"><?php echo $product['description']; ?></p>
+                            <div class="product-bodyend">
```
proc tady nejde koupit zbozi?

AsemikTEA replied:
vyřešeno

nvbach91 commented on www/talp03/sp/Model/ProductDB.php:
```diff
@@ -0,0 +1,91 @@
+    public function findAll() {
+        $statement = $this->pdo->prepare('SELECT * FROM products');
+        $statement->execute();
+        return $statement->fetchAll(PDO::FETCH_ASSOC);
```
to PDO::FETCH_ASSOC nastavte pomoci atributu pri vytvareni instance pripojeni

AsemikTEA replied:
vyřešeno

nvbach91 reviewed (changes requested):
- jak se dostanu do administrace?

AsemikTEA committed:
Review changes

přesunutí informací k připojení do DB do .env, Změna uložení ceny objednávky místo GET do SESSION, nastvení default modu pro fetch u inicializace pdo, umožnění nákupu produktu přímo z wishlistu

AsemikTEA commented:
Dal jsem k vašemu účtu privilegium, teď byste se měl mít přístup k funkcím admina. 
Ty složky v kořenu projektu jsem tam nahrál omylem, měl jsem vzít normálně tu www složku ale vzal jsem jen ty soubory vevnitř.

nvbach91 commented:
- URL proc jsou tam vsude View? htaccess?
- wiishlist error kdyz je prazdny
- presmerovani po dokonceni objednavky
- zruseni s potvrzeni
- neprihlaseni uzivatel nenakoupi
- category select nazev
- id needitovatelny
- privilegium jakekoliv cislo?
- stranka pro admina aby vyrizoval objednavky
- ukazka databaze

AsemikTEA committed:
Obhajoba

nvbach91 committed:
remove invalid folders
